server-core/engine/src/main/java/io/infectnet/server/engine/core/script/generation/ScriptGeneratorImpl.java

```diff
@@ -37,6 +37,8 @@ private ScriptGeneratorImpl(Set<CompilationCustomizer> customizers) {
   @Override
   public Script generateFromCode(String sourceCode) throws ScriptGenerationFailedException {
     try {
+      logger.debug("Compiling new source: {} ...", sourceCode.substring(0, 15));
```

Comment:
Remove the hard-coded 15 value and extract it into a constant.

This can cause exceptions, so please use `max(LENGTH, sourceCode.length())` or use `StringUtils.left()` which does not throw exceptions.